### ISSUE TITLE
"OutOfBoundsException" when retrieving "Location" custom data from "QBDialog"

### ISSUE THREAD

**Issue by regmoraes** [QuickBlox/quickblox-android-sdk-releases#3]
I have a method that creates a "QBDialog" and attaches a "Location" custom data to it:

```
List<Double> myLocation = new ArrayList<>();
        myLocation.add(locationManager.getLastLocation().getLatitude());
        myLocation.add(locationManager.getLastLocation().getLongitude());

        QBDialogCustomData data = new QBDialogCustomData("JsRoom");
        data.putLocation("room_location", myLocation);

        final QBDialog qbDialog = new QBDialog();
        qbDialog.setName("Test");
        qbDialog.setType(QBDialogType.PUBLIC_GROUP);
        qbDialog.setCustomData(data);

        try {

            subscriber.onNext(
                    QBChatService
                            .getInstance()
                            .getGroupChatManager()
                            .createDialog(qbDialog));

            subscriber.onCompleted();

        } catch (QBResponseException e) {

            subscriber.onError(e);
        }
```

But when I try to retrieve the Location by calling `room.getCustomData().getLocation("room_location")` I get this error even if the `getLocation()` returns a non null response:

```
java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
 at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
 at java.util.ArrayList.set(ArrayList.java:481)
 at com.quickblox.core.model.QBBaseCustomObject.getLocation(QBBaseCustomObject.java:291)
```

Maybe it's something wrong with the Quickblox SDK during creation of the Array that returns the location values.

**Comment by soulfly** [comment]
Could you please also post here REST logs from logcat
We need to check what you are getting from the backend in a response

**Comment by soulfly** [comment]
Hi there, 

can you try this patch
https://sdk.quickblox.com/android/artifacts/artifacts_customdata_location_fix.zip

we found one strange thing in the behaviour and made the patch

**Comment by soulfly** [comment]
Fixed with Android 2.5.2
http://quickblox.com/developers/Android#Framework_changelog
